### ISSUE TITLE
Unable to reproduce the results

### ISSUE THREAD

**Issue by westfish** [Shark-NLP/DiffuSeq#22]
Unable to reproduce the results of the question generation task in the paper.

- The results of the question generation task in the paper are as follows：
<img width="405" alt="image" src="https://user-images.githubusercontent.com/20476674/212636002-81e5be83-897d-4dbb-bd2b-baffda96e671.png">
-We use the provided command to reproduce the following results：
![image](https://user-images.githubusercontent.com/20476674/212636594-bae4b9ee-4d37-45ca-b720-a0b866268e09.png)
The command provided by repo is also the command we use, as follows：
```python
python -m torch.distributed.launch --nproc_per_node=4 --master_port=12233 --use_env run_train.py --diff_steps 2000 --lr 0.0001 --learning_steps 40000 --save_interval 2000 --seed 102 --noise_schedule sqrt --hidden_dim 128 --bsz 2048 --microbatch 64 --dataset qg --data_dir {datasets/QG} --vocab bert --seq_len 128 --schedule_sampler lossaware --notes qg
```

**Comment by summmeer** [comment]
Hi,
Do you try to generate several candidates using different random seeds and evaluate them using MBR? In our paper, the size of the candidate set $|S|=10$.

**Comment by westfish** [comment]
It works! Results improved after using MBR.

![image](https://user-images.githubusercontent.com/20476674/215694541-ba4c7ad1-a228-494c-b5b9-c9684957a22c.png)

**Comment by scvready123** [comment]
@westfish Hi,I wonder which version of your "nltk" sitepackage is when you reproduce the results.I firstly use nltk==3.5,and the results is close to the paper.But when I use nltk==3.6.7, the BLEU score is far less than the former.I have read the source codes of nltk, but I can't figure it out.It's shocking!

**Comment by zzbuzzard** [comment]
Hi! Using MBR with 10 samples on the paraphrasing task with the pre-trained checkpoint, I obtained the following:

BLEU:   0.2179  (paper has 0.2413)
ROUGE-l:   0.5743  (paper has 0.5880)
berscore:   0.8120  (paper has 0.8365)
dist1:   0.9650  (paper has 0.9807)

i.e. similar results but consistently lower.

I wonder if it's because I used `run_decode_solver` (DPM++) rather than `run_decode` for decoding? Unfortunately running without DPM++ 10 times is probably too expensive for me to check.

Edit: it's probably because I was using DPM++ with just 10 steps (the default). I will try again with more steps!

**Comment by summmeer** [comment]
@zzbuzzard Yeah, you're right. In our DiffuSeq-v2 paper (Figure 3), you will find that using DPM-solver++ would affect the BLEU score, but the situation is much better than using DDIM.
